pkg/validation/validate/schema_props.go

```diff
@@ -41,6 +41,18 @@ type schemaPropsValidator struct {
 
 func (s *schemaPropsValidator) SetPath(path string) {
 	s.Path = path
+	for _, v := range s.anyOfValidators {
+		v.SetPath(path)
+	}
+	for _, v := range s.allOfValidators {
+		v.SetPath(path)
+	}
+	for _, v := range s.oneOfValidators {
+		v.SetPath(path)
+	}
+	if s.notValidator != nil {
+		s.notValidator.SetPath(path)
+	}
```

Comment:
Note for onlookers: we would not panic on calling `SetPath()` on a `nil` `notValidator` except for the fact that `SetPath()` accesses the `Path` struct member and does not check if the receiver (i.e. itself) is `nil`, which would cause a panic. Therefore, we check that `notValidator` is not `nil` before calling `SetPath()`.